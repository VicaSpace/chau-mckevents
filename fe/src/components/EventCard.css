.event-card {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 0px;

    width: 832.95px;
    height: 198px;

    background: #FFFFFF;
    border: 1px solid rgba(0, 0, 0, 0.25);
    border-radius: 10px;

    margin-top: 25px;
    flex: 1;
}

.event-date {
    width: 154.72px;
    height: 57.12px;
    margin-top: 11.53px;
    margin-left: 30.36px;
}

.event-date-month {
    font-family: 'Inter';
    font-style: normal;
    font-weight: 600;
    font-size: 32px;
    line-height: 39px;

    color: #263238;
    text-align: center;
}

.event-year {
    text-align: center;

    font-family: 'Inter';
    font-style: normal;
    font-weight: 600;
    font-size: 15px;
    line-height: 18px;
}

.event-time {
    width: 301px;
    height: 14px;
    left: 254px;
    margin-top: 0px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 14px;
    /* identical to box height */

    color: #000000;
}

.event-location {
    width: 336px;
    height: 42px;
    left: 255px;
    margin-top: 5px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 14px;

    color: #000000;
}

.event-content {
    display: flex;
}