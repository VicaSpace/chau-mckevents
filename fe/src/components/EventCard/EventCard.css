.event-card {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 0px;

    width: 832.95px;
    height: 198px;

    background: #FFFFFF;
    border: 1px solid rgba(0, 0, 0, 0.25);
    border-radius: 10px;

    margin-top: 25px;
    flex: 1;
}
