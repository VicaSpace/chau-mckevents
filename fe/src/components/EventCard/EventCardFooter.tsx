import {
  CheckCircleOutlined,
  EllipsisOutlined,
  ShareAltOutlined,
} from '@ant-design/icons';
import React from 'react';
import './EventCardFooter.css';

const EventCardFooter: React.FC<{ status: string }> = ({ status }) => {
  return (
    <div className="card-footer" data-testid="test-card-footer">
      <div className="card-footer-item">
        <CheckCircleOutlined
          data-testid="check-icon"
          style={status === 'CONFIRMED' ? { color: 'green' } : {}}
        />
      </div>

      <div className="card-footer-item">
        <ShareAltOutlined />
      </div>

      <div className="card-footer-item">
        <EllipsisOutlined />
      </div>
    </div>
  );
};

export default EventCardFooter;
