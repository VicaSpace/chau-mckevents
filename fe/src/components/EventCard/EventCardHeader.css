.card-head {
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    padding: 0px 24px;

    width: 832.95px;
    height: 56px;

    background: #FFFFFF;
}


.event-name {
    width: 148px;
    height: 24px;

    margin-top: 16px;
    margin-left: 24px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;

    color: rgba(0, 0, 0, 0.85);
}


.event-organizer {
    position: absolute;
    width: 247px;
    height: 20px;
    margin-left: 254px;
    margin-top: 36px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 19px;

    color: #000000;
    
}

.event-office {
    position: absolute;
    width: 117px;
    height: 22px;

    margin-left: 691.95px;
    margin-top: 17px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 22px;

    color: #1890FF;
}