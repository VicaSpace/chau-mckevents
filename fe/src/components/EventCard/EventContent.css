
.event-date {
    width: 154.72px;
    height: 57.12px;
    margin-top: 11.53px;
    margin-left: 30.36px;
}

.event-date-month {
    font-family: 'Inter';
    font-style: normal;
    font-weight: 600;
    font-size: 32px;
    line-height: 39px;

    color: #263238;
    text-align: center;
}

.event-year {
    text-align: center;

    font-family: 'Inter';
    font-style: normal;
    font-weight: 600;
    font-size: 15px;
    line-height: 18px;
}

.event-time {
    width: 301px;
    height: 14px;
    left: 254px;
    margin-top: 0px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 14px;
    /* identical to box height */

    color: #000000;
    position: relative;
}

.event-location {
    width: 336px;
    height: 42px;
    left: 255px;
    margin-top: 5px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 14px;

    color: #000000;
    position: relative;
}

.event-content {
    display: flex;
    flex-direction: row;

    position: absolute;
}