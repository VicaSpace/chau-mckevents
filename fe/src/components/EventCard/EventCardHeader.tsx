import React from 'react';
import './EventCardHeader.css';

const EventCardHeader: React.FC<{
  name: string;
  organizer: string;
  office: string;
}> = ({ name, organizer, office }) => {
  return (
    <div className="card-head">
      <div className="event-name">{name}</div>
      <div className="event-organizer">Organizer: {organizer}</div>
      <div className="event-office">{office}</div>
    </div>
  );
};

export default EventCardHeader;
