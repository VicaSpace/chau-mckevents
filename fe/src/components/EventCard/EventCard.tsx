import React from 'react';
import './EventCard.css';
import EventCardFooter from './EventCardFooter';
import EventCardHeader from './EventCardHeader';
import EventContent from './EventContent';

interface EventCardProps {
  name: string;
  date: string;
  organizer: string;
  time: string;
  location: string;
  office: string;
  timeSuggestions: any[];
  status: string;
}

const EventCard: React.FC<EventCardProps> = ({
  name,
  date,
  organizer,
  time,
  location,
  office,
  timeSuggestions,
  status,
}) => {
  return (
    <div className="event-card">
      <EventCardHeader name={name} organizer={organizer} office={office} />
      <EventContent
        date={date}
        time={time}
        timeSuggestions={timeSuggestions}
        location={location}
        status={status}
      />
      <EventCardFooter status={status} />
    </div>
  );
};

export default EventCard;
