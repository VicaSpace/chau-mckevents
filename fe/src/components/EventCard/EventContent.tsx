import moment from 'moment';
import React from 'react';
import './EventContent.css';

const EventContent: React.FC<{
  date: string;
  time: string;
  location: string;
  timeSuggestions: any[];
  status: string;
}> = ({ date, time, location, timeSuggestions, status }) => {
  const renderTime = () => {
    if (status === 'CONFIRMED') {
      return <>Time: {moment(time).format('HH:MM').toString()}</>;
    } else if (status === 'NOT_CONFIRMED') {
      return (
        <>
          Time suggestions:{' '}
          {timeSuggestions
            .map((suggestion) =>
              moment(suggestion.time).format('HH:MM').toString()
            )
            .join(' / ')}
        </>
      );
    }
  };

  return (
    <div className="card-content" data-testid="test-card-content">
      <div className="event-date">
        <div className="event-date-month">
          {moment(date).format('MMMM Do').toString()}
        </div>
        <div className="event-year">
          {moment(date).format('YYYY').toString()}
        </div>
      </div>
      <div className="event-time" data-testid="test-event-time">
        {renderTime()}
      </div>
      <div className="event-location" data-testid="test-event-location">
        Location: {location}
      </div>
    </div>
  );
};

export default EventContent;
