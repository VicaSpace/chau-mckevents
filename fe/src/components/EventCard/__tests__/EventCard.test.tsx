import {render, screen} from '@testing-library/react'
import React from 'react'
import EventCard from '../EventCard';

const dummyProps = {
    name: 'test',
    date: '2020-02-02 00:00:00',
    organizer: 'test-organizer',
    time: '2020-02-02 08:00:00',
    location: 'test',
    office: 'test',
    timeSuggestions: [
        {
            id: 1,
            time: '2020-02-02 08:00:00',
        },
        {
            id: 1,
            time: '2020-02-02 09:00:00',
        },
    ],
    status: 'CONFIRMED'
}

describe('Test: Event Card component screenshot', () => {
    it('should render the Event Card correctly', () => {
      const { container } = render(
          <EventCard {...dummyProps} status='NOT_CONFIRMED' />
      );
      expect(container).toMatchSnapshot();
    });
  });

describe('Test: Event Card node', () => {
    it('should render header node', () => {
        render(<EventCard {...dummyProps} status='NOT_CONFIRMED' />);
        const eventLocation = screen.getByTestId('test-card-head');
        expect(eventLocation.classList).toContain('card-head');
    });

    it('should render content node', () => {
        render(<EventCard {...dummyProps} status='NOT_CONFIRMED' />);
        const eventLocation = screen.getByTestId('test-card-content');
        expect(eventLocation.classList).toContain('card-content');
    });

    it('should render footer node', () => {
        render(<EventCard {...dummyProps} status='NOT_CONFIRMED' />);
        const eventLocation = screen.getByTestId('test-card-footer');
        expect(eventLocation.classList).toContain('card-footer');
    });
});
