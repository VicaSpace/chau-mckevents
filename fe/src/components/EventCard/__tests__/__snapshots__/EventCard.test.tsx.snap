// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Event Card component screenshot should render the Event Card correctly 1`] = `
<div>
  <div
    class="event-card"
  >
    <div
      class="card-head"
    >
      <div
        class="event-name"
      >
        test
      </div>
      <div
        class="event-organizer"
      >
        Organizer: 
        test-organizer
      </div>
      <div
        class="event-office"
      >
        test
      </div>
    </div>
    <div
      class="card-content"
    >
      <div
        class="event-date"
      >
        <div
          class="event-date-month"
        >
          February 2nd
        </div>
        <div
          class="event-year"
        >
          2020
        </div>
      </div>
      <div
        class="event-time"
        data-testid="test-event-time"
      >
        Time suggestions:
         
        08:02 / 09:02
      </div>
      <div
        class="event-location"
        data-testid="test-event-location"
      >
        Location: 
        test
      </div>
    </div>
    <div
      class="card-footer"
    >
      <div
        class="card-footer-item"
      >
        <span
          aria-label="check-circle"
          class="anticon anticon-check-circle"
          data-testid="check-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="check-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
      </div>
      <div
        class="card-footer-item"
      >
        <span
          aria-label="share-alt"
          class="anticon anticon-share-alt"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="share-alt"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 000-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
            />
          </svg>
        </span>
      </div>
      <div
        class="card-footer-item"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
